File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    m=0
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.helper(root)
        return self.m
    def helper(self ,root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        l=self.helper(root.left)
        r=self.helper(root.right)
        self.m=max(l+r,self.m)
        return max(l,r) + 1

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int p;
int ans;
    int kthSmallest(TreeNode* root, int k) {
      p=k;
      helper(root);
      return ans;
    }

    void helper(TreeNode* root){
        if(!root || p==0)
            return;
        helper(root->left);
        p--;
        if(p==0)
        {
            ans=root->val;
            return;
        }
        helper(root->right);
    }
};